### DISCUSSION TITLE
How do I forward command line arguments when debugging into a dotnet console program?

### DISCUSSION THREAD

**Discussion by Amplus** [microsoft/vscode-docker#4256] [Q&A] [answered]
Greetings,

I am trying to pass command line arguments into a C# dotnet8 console application and I haven't been able to figure it out. I would think the appropriate way would be to inject them via the generated docker-run task in tasks.json using one of the documented properties here:
https://code.visualstudio.com/docs/containers/reference#_run-task-reference
I have spent a lot of time trying to get it to work to with no success. Any pointers would be greatly appreciated. 

```
{
            "type": "docker-run",
            "label": "docker-run: debug",
            "dependsOn": [
                "docker-build: debug"
            ],
            "dockerRun": {
               //Which property can I use here to add command line args that will populate the args array inside my console app
               //
            },
            "netCore": {
                "appProject": "${workspaceFolder}/MyContainerizedConsoleApp.csproj",
                "enableDebugging": true
            }
        }
}
```
I have noticed that in the provided dockerfile that the ENTRYPOINT line doesn't seem to have any effect or adding CMD statements there.

```
FROM mcr.microsoft.com/dotnet/runtime:8.0 AS base
WORKDIR /app

USER app
FROM --platform=$BUILDPLATFORM mcr.microsoft.com/dotnet/sdk:8.0 AS build
ARG configuration=Release
WORKDIR /src
COPY ["MyContainerizedConsoleApp.csproj", "./"]
RUN dotnet restore "MyContainerizedConsoleApp.csproj"
COPY . .
WORKDIR "/src/."
RUN dotnet build "MyContainerizedConsoleApp.csproj" -c $configuration -o /app/build

FROM build AS publish
ARG configuration=Release
RUN dotnet publish "MyContainerizedConsoleApp.csproj" -c $configuration -o /app/publish /p:UseAppHost=false

FROM base AS final
WORKDIR /app
COPY --from=publish /app/publish .
ENTRYPOINT ["dotnet", "MyContainerizedConsoleApp.dll"]
```
Again any pointers are greatly appreciated, I am running this on OSX Sonoma Apple M3. I am aware that I can use the env argument inside dockerRun but I want to also be able to specify command line arguments.

**Comment by bwateratmsft** [comment, marked as answer]
For debugging, we don't actually launch the entrypoint--we start the container with a do-nothing command so it stays up but isn't doing anything. From there, we start VSDBG, and instruct VSDBG to start your app. This way, you can debug startup code.

Consequently, to get command line arguments to your app, you have to instruct VSDBG to do it, which can be done in `.vscode/launch.json` by adding properties like this to the "Docker .NET Launch" configuration:

```jsonc
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Docker .NET Launch",
            "type": "docker",
            "request": "launch",
            "preLaunchTask": "docker-run: debug",
            "netCore": {
                "appProject": "${workspaceFolder}/NetConsole8.csproj"
            },
            "args": [ // Add this section here
                "/app/bin/Debug/net8.0/NetConsole8.dll", // The path to your app will be something like /app/bin/Debug/net8.0/YourAppName.dll
                "-Hello", // First argument
                "-World" // Second argument
            ]
        }
    ]
}
```

The `"args"` property will get a yellow warning squiggle saying it is not valid, but you can ignore that.

**Reply by Amplus** [reply]
Thanks a lot @bwateratmsft this was not obvious to me, I was sticking the args array inside the "netCore" section.
